[23기 김동우] ADD : [Mission 6] 좋아요 기능 작성

Hussbogo opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 미션 1, 2, 3, 4, 5, 9 완료
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)

(commit 860992d - Mission 1, 2)
- 로그인 ID/Password 유무 (글자수 제한은 없음) 작성
- 로그인 버튼, ID, Password 상호작용 작성
- 로그인 버튼 클릭 ID 정보 localstorage 저장 작성

(commit fa105c2 - Mission 3, 4, 9)
- 메인 레이아웃 작성
- 메인 댓글 id, localstorage id value 할당 작성
- 메인 댓글 추가 기능 작성
- 메인 엔터키, 게시 버튼 이벤트 작성
- viewport-width 값에 따른 레이아웃 변경 작성 (로그인 / 메인)

(commit 41cfcbe - Mission 5)
- 로그인 ID, Password 유효성 검증 작성
- 로그인 form 제출 후 메인 전환 개선

(commit aac376f - Modify : localStrage 전달방식 개선)
- 기존 click or Enter press evnet- > form submit  

(commit 0696544 - Mission 1,3 미비점 보완)
- 로그인 footer div 내용 작성
- 메인 recommended-block 하단 nav 작성

(commit 0f8e9f1 - Modify : 댓글정보 저장방식 수정)
- 기존 button, input -> form submit 으로 변경
- localStorage 내부 데이터 구조 변경 (String 단순 저장 -> Array)
- load / save 기능 로직 수정

(commit 8a54cae - ADD : Mission 6 댓글 좋아요 기능 구현)
- 댓글 좋아요 기능 작성
- 댓글 삭제 기능 작성 예정

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 많이 배워서 원활하게 프로젝트를 진행하고 싶습니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [X] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

saengmotmi commented on students/10th/yeajinpark/login.html:
```diff
@@ -0,0 +1,26 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <title>repl.it</title>
+    <link href="style/common.css" rel="stylesheet" type="text/css" />
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
```
```suggestion
    <link href="style/common.css" rel="stylesheet" type="text/css" />
    <link href="style/login.css" rel="stylesheet" type="text/css" />
```

indent 한칸 앞으로 당겨서 다른 요소들과 일치시켜주세요!

igobyyj replied:
넵!

saengmotmi commented on students/10th/yeajinpark/login.html:
```diff
@@ -0,0 +1,26 @@
+  </head>
+  <body>
+      <div class = "background" >
+        <img src = "img/logo_text.png">
+
+        <form class = "input-wrap">
```
```suggestion
        <img src="img/logo_text.png">
        <form class="input-wrap">
```

불필요한 줄바꿈은 없애주세요. 속성과 속성값 사이의 공백도 없애주시는게 좋겠습니다.

igobyyj replied:
넵 공백 딜리트하겠습니다.

saengmotmi commented on students/10th/yeajinpark/login.html:
```diff
@@ -0,0 +1,26 @@
+        <form class = "input-wrap">
+          <input class = "userinfo" type = "text" placeholder = "전화번호, 사용자 이름 또는 이메일">
+          <input class = "userinfo" type = "password" placeholder = "비밀번호">
+          <button> 로그인 </button>
```
```suggestion
          <button>로그인</button>
```

불필요한 공백이 들어 있네요. 사소해보이지만 이런 습관 하나 하나가 좋은 코드를 작성하는데 도움이 됩니다 :)

igobyyj replied:
넵 공백 딜리트하겠습니다.

saengmotmi commented on students/10th/yeajinpark/login.html:
```diff
@@ -0,0 +1,26 @@
+    <link href="style/common.css" rel="stylesheet" type="text/css" />
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+      <div class = "background" >
+        <img src = "img/logo_text.png">
```
```suggestion
  <body>
    <div class = "background" >
      <img src = "img/logo_text.png">
```

부모 자식 태그 간의 관계는 띄어쓰기 두 칸으로 표기해주세요.

참고로 vscode를 사용하신다면 Ctrl + K, F(windows), Shift + Option + F(Mac OS)로 자동 줄맞춤을 해줄 수 있습니다.

igobyyj replied:
꿀팁 감쟈합니다.

joonsikyang commented on students/10th/yeajinpark/westagram/js/login.js:
```diff
@@ -0,0 +1,12 @@
+document.addEventListener("keyup", function() {
+    const pw = document.getElementById('id').value;
+    const id = document.getElementById('pw').value;
+
+    if (pw !== "" && id!=="") {
+        const loginButton = document.getElementsByClassName('login-btn')[0];
+        loginButton.style.color = '#3e95ef';
+    }
+});
```
다시 입력값을 지우면 버튼 색이 되돌아오나요? 확인 후 댓글로 알려주세요 :)

igobyyj replied:
넵, 확인하고 다시 올리겠습니다.

joonsikyang commented on students/10th/yeajinpark/westagram/guide/events.md:
```diff
@@ -0,0 +1,10 @@
+자바스크립트 이벤트를 구현해주세요!
+
+# 로그인 페이지
+
+1. id, pw 에 각각 한 글자 이상 써야 버튼이 활성화 되도록 해주세요.
+- 원래 연한 파란색이었다가 -> 활성화 되면 파란색으로!
+
+# 메인 페이지
+1. 댓글 input 창에 엔터치거나 "게시" 누르면 댓글 추가되도록
+- createElement로 요소 생성해서, input에 입력한 값이 추가 되어야합니다.
```
guide 폴더는 지워주세요!

igobyyj replied:
넵, 지우고 다시 올리겠습니다.

dongwu-hugo committed:
ADD : 로그인 레이아웃 / 기능 작성
- 로그인 레이아웃 작성
- 로그인 ID/Password 유무 작성
- 로그인 버튼, ID, Password 상호작용 작성
- 로그인 버튼 클릭 ID 정보 localstorage 저장 작성

dongwu-hugo committed:
ADD : Mission 3, 4 완료
- 메인 레이아웃 작성
- 메인 댓글 추가 게시버튼 or  엔터키 작성

dongwu-hugo committed:
ADD : Mission 5 complete
- ID Email, Password 8자 유효성 검증 작성

dongwu-hugo committed:
MODIFY : localstorage 저장 및 정보전달방식 개선
- 기존 click event, keyup event form 전달로 개선

dongwu-hugo committed:
ADD : 로그인/메인 레이아웃 footer
- 로그인 footer div 내용 작성
- 메인 recommended-block 하단 nav 작성

dongwu-hugo committed:
MODIFY : 댓글정보 저장방식 수정
- 기존 id, text 1:1 대응 방식 -> 한 번에 처리
- button, input.value -> form 변경
- localStorage 내부 데이터 구조 변경

dongwu-hugo committed:
MODIFY : Code Refactoring
- HTML, CSS, Js Refactoring

dongwu-hugo committed:
ADD : 상단 nav modal 작성
- header 상단 modal on/off 작성

dongwu-hugo committed:
ADD : 댓글 좋아요 기능 작성 / 삭제 추가 예정
- 미션 6 좋아요 기능 구현
- 미션 6 삭제 기능 예정

dongwu-hugo committed:
MODIFY : 댓글 좋아요 기능 작성
- 기존 방식 : addListener  콜백함수 호출 -> 클로저로 인자 전달 변경
- 그 외 코드 Refactoring
